File: 프로그래머스/0/120866. 안전지대/안전지대.java
class Solution {
    private int[] dy = {-1, -1, -1, 0, 1, 1, 1, 0};
    private int[] dx = {-1, 0, 1, 1, 1, 0, -1, -1};
    public int solution(int[][] board) {
        int ret = 0;
        int n = board.length;
        int m = board[0].length;
        for(int i = 0; i < n; i++){
            for(int j = 0; j < m; j++){
                if(board[i][j] == 0 || board[i][j] == -1) continue;
                for(int d = 0; d < 8; d++){
                    int ny = i + dy[d];
                    int nx = j + dx[d];
                    if(ny < 0 || nx < 0 || ny >= n || nx >= m || board[ny][nx] == 1) continue;
                    board[ny][nx] = -1;
                }
            }
        }
        for(int i = 0; i < n; i++){
            for(int j = 0; j < m; j++){
                if(board[i][j] == 0) ret++;
            }
        }
        return ret;
    }
}

File: 프로그래머스/0/120866. 안전지대/안전지대.py
def solution(board):
    dy = [-1, 0, 1, 0, 1, 1, -1,-1]
    dx = [0, 1, 0, -1, -1,1, 1, -1]
    n, m = len(board), len(board[0])
    temp = [[0]*m for _ in range(n)]
    for i in range(n):
        for j in range(m):
            for d in range(8):
                if board[i][j] == 0: continue
                temp[i][j] = 1
                ny, nx = i + dy[d], j + dx[d]
                if ny < 0 or nx < 0 or ny >= n or nx >= m:
                    continue
                temp[ny][nx] = 1
    ret = 0
    for i in range(n):
        ret += sum(1 for j in range(m) if temp[i][j] == 0)
    return ret